step-framework-timeseries/src/main/java/step/core/timeseries/ingestion/TimeSeriesIngestionPipeline.java

```diff
@@ -116,19 +118,26 @@ public void ingestPoint(Map<String, Object> attributes, long timestamp, long val
 
     /**
      * Flush will be triggered only on the specified ingestion. Other pipelines in the chain will not be flushed.
+     * This method forces the flush independently of the resolution or max queue size, it should be called for instance
+     * once an execution completed.
      */
     public void flush() {
         flush(true);
     }
 
-    private void flush(boolean forceFlush) {
+    /**
+     *  Flush is performed only for the current ingestion pipeline (not the nested ones)
+     *  It's usually only called directly by the schedule jobs flushing the time series on regular basis, it is also directly called for some Junit tests (located in different package)
+     * @param forceFlush if false, the flush is only performed is the time interval is complete (i.e. bigger than the resolution) or if the queue size limit is reached
+     */
+    public void flush(boolean forceFlush) {
         lock.writeLock().lock();
         try {
             trace("Flushing");
             long now = System.currentTimeMillis();
 
             seriesQueue.forEach((k, v) -> {
-                if (forceFlush || k + sourceResolution < now - FLUSH_OFFSET) {
+                if (forceFlush || ((k + sourceResolution) < (now - FLUSH_OFFSET)) || seriesQueue.get(k).size() >= seriesQueueSizeflushThreshold) {
```

Comment:
The seriesQueue should be available in the variable "v". Not necessary to retrieve it with get(k)